### ISSUE TITLE
telepresence container method needs to check for socat

### ISSUE THREAD

**Issue by rhs** [telepresenceio/telepresence#336]
### What were you trying to do?

telepresence --docker-run -it ubuntu

### What did you expect to happen?

it worky

### What happened instead?

it no worky

### Automatically included information

Command line: `['cli/telepresence', '--docker-run', '-it', 'ubuntu']`
Version: `0.68`
Python version: `3.6.2 (default, Sep 22 2017, 08:28:09) 
[GCC 7.2.1 20170915 (Red Hat 7.2.1-2)]`
kubectl version: `Client Version: v1.8.0`
oc version: `(error: [Errno 2] No such file or directory: 'oc')`
OS: `Linux venture 4.12.14-300.fc26.x86_64 #1 SMP Wed Sep 20 16:28:07 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux`
Traceback:

```
Traceback (most recent call last):
  File "cli/telepresence", line 257, in call_f
    return f(*args, **kwargs)
  File "cli/telepresence", line 2370, in go
    runner, args
  File "cli/telepresence", line 1536, in start_proxy
    processes, socks_port, ssh = connect(runner, remote_info, args)
  File "cli/telepresence", line 1171, in connect
    ), "TCP4:127.0.0.1:{}".format(ssh.port)
  File "cli/telepresence", line 404, in popen
    return self.launch_command(track, *args, **kwargs)
  File "cli/telepresence", line 358, in launch_command
    process = Popen(*args, **kwargs)
  File "/usr/lib64/python3.6/subprocess.py", line 707, in __init__
    restore_signals, start_new_session)
  File "/usr/lib64/python3.6/subprocess.py", line 1333, in _execute_child
    raise child_exception_type(errno_num, err_msg)
FileNotFoundError: [Errno 2] No such file or directory: 'socat'

```

Logs:

```
sence-1508984557-6758008-5423-440061768-vgmhk', '-o', 'json'],)...
   4.5 TL | [9] captured.
   4.7 TL | [10] Capturing: (['kubectl', '--context', 'gke_datawireio_us-central1-a_rhs', '--namespace', 'default', 'get', 'pod', 'telepresence-1508984557-6758008-5423-440061768-vgmhk', '-o', 'json'],)...
   5.1 TL | [10] captured.
   5.1 TL | [11] Launching: (['kubectl', '--context', 'gke_datawireio_us-central1-a_rhs', '--namespace', 'default', 'logs', '-f', 'telepresence-1508984557-6758008-5423-440061768-vgmhk', '--container', 'telepresence-1508984557-6758008-5423'],)...
   5.1 TL | [12] Launching: (['kubectl', '--context', 'gke_datawireio_us-central1-a_rhs', '--namespace', 'default', 'port-forward', 'telepresence-1508984557-6758008-5423-440061768-vgmhk', '46673:8022'],)...
   5.1 TL | [13] Capturing: (['ip', 'addr', 'show', 'dev', 'docker0'],)...
   5.1 TL | [13] captured.
   5.1 TL | [14] Launching: (['socat', 'TCP4-LISTEN:46673,bind=172.17.0.1,reuseaddr,fork', 'TCP4:127.0.0.1:46673'],)...

```

**Comment by plombardi89** [comment]
Possibly related to this? https://github.com/datawire/telepresence/issues/278

**Comment by rhs** [comment]
This works if I do a dnf install socat. I think the code is just missing a check/useful error message when socat is not installed.

**Comment by ark3** [comment]
Fixed in #720.
